### ISSUE TITLE
stackoverflow in integral

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#954]
Imported from SourceForge on 2024-07-03 15:46:51
Created by **[kcrisman](https://sourceforge.net/u/kcrisman/)** on 2016-04-20 00:22:59
Original: https://sourceforge.net/p/maxima/bugs/3144

---

```
Maxima 5.36.1 http://maxima.sourceforge.net
using Lisp SBCL 1.2.11-1.fc22
(%i1) integrate(sin(k*x)/x*erf(x^2),x,0,inf);
Maxima encountered a Lisp error:
Control stack exhausted (no more space for function call frames).
```
See http://trac.sagemath.org/ticket/20467 as well as http://ask.sagemath.org/question/33144/unable-to-make-sense-of-maxima-expression-as-result-of-integration/ which has a different error which however I couldn't track down and may be unrelated to Maxima.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 15:46:53
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-04-23 06:15:11
Original: https://sourceforge.net/p/maxima/bugs/3144/#4d5b

---

This seems to be a mistaken recursion in PTIMES%E which wants to call POLFACTORS to split up an expression and then try again on the result. But if POLFACTORS can't actually factor it, PTIMES%E is called again with the same argument, leading to a stack overflow.

I think I have a bug fix to prevent the stack overflow, although it doesn't help Maxima find the integral. Stay tuned.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 15:46:56
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-04-24 05:32:12
Original: https://sourceforge.net/p/maxima/bugs/3144/#77b1

---

- **labels**:  --> integrate, defint
- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 15:47:00
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-04-24 05:32:12
Original: https://sourceforge.net/p/maxima/bugs/3144/#3fff

---

Fixed by commit 4568937. Closing this report.

With this bug fix, Maxima can't find the integral, but at least it just returns a noun expression, instead of crashing.
